dist/categorical.go

```diff
@@ -0,0 +1,177 @@
+// Copyright ©2015 The gonum Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package dist
+
+import (
+	"math"
+	"math/rand"
+)
+
+// Categorical is an extension of the Bernouilli distribution where x can take
+// a certain number of possible values. Categorical must be initialized with
```

Comment:
If I understand it correctly, the possible values are fixed to be `{0, 1, ..., k = len(w)-1}`. Is there any reason not to say that? Especially because you talk about zero-indexedness on the line below?